### ISSUE TITLE
[actions] if actions use Re-run all jobs to start  workflows,it will ignore "jobs.<job_id>.needs"

### ISSUE THREAD

**Issue by zoy0** [go-gitea/gitea#25925]
### Description

I use re-run all jobs to rerun a workflow, but instead of sequential, alljobs are executed concurrently.This leads to needs. <job>.outputs cannot be used.
```
name: outputs
on: push

jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - name: step1
        run: |
          echo "first job1 and first step"          

  job2:
    runs-on: ubuntu-latest
    outputs:
      output1: ${{ steps.step1.outputs.output1 }}
      output2: ${{ steps.step2.outputs.output2 }}
    steps:
      - name: step1
        id: step1
        run: |
          date -Is > output1
          cat output1
          echo "output1=$(cat output1)" >> $GITHUB_OUTPUT                    
      - name: step2
        id: step2
        run: |
          cat /proc/sys/kernel/random/uuid > output2
          cat output2
          echo "output2=$(cat output2)" >> $GITHUB_OUTPUT           

  job3:
    needs: [job1,job2]
    runs-on: ubuntu-latest
    steps:
      - run: echo ${{ needs.job2.outputs.output1 }}
      - run: echo ${{ needs.job2.outputs.output2 }}
      - run: echo ${{ needs.job2.result }}
```

normal operation
![image](https://github.com/go-gitea/gitea/assets/105140381/a6840654-f931-4d75-a0c1-1ffae25d1132)

re-run all jobs:
![image](https://github.com/go-gitea/gitea/assets/105140381/e9f178a7-c209-4648-a849-6186689b041d)

### Gitea Version

1.20

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Linux

### How are you running Gitea?

docker images:  gitea/gitea:1.20  and  gitea/act_runner:0.2.3

### Database

SQLite

**Comment by wolfogre** [comment]
Earlier, it was not supported to rerun all jobs, you could only rerun the jobs one by one.

After #24350, Gitea provided a batch operation to rerun all jobs, unfortunately, it was just a batch operation which means help to rerun the jobs one by one. Reconstructing dependencies of jobs may not be easy, before it has been resolved, you could do it manually.

<img width="424" alt="image" src="https://github.com/go-gitea/gitea/assets/9418365/c7c2d646-71cd-4ab3-99b6-cb8739697884">

**Comment by Zettat123** [comment]
Closed by #29955
